app/controllers/RiderStageConnectionController.java

```diff
@@ -66,19 +65,15 @@ public RiderStageConnectionController(StageRepository stageRepository, RiderRepo
     }
 
     @BodyParser.Of(BodyParser.Json.class)
-    public CompletableFuture<Result> addStage() {
+    public CompletionStage<Result> updateRiderStageConnection(long riderStageConnectionId) {
         JsonNode json = request().body().asJson();
-        return parseStage(json).thenApply(stage -> stageRepository.addStage(stage)).thenApply(message -> {
-            return ok(message + " has been added");
+        return parseRiderStageConnection(json, riderStageConnectionId).thenApply(riderStageConnectionRepository::updateRiderStageConnection).thenApplyAsync(rSC -> {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove useless curly braces around statement and then remove useless return keyword (sonar.java.source not set. Assuming 8 or greater.) [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1602)